packages/ui-components/src/components/dropdown-base/dropdown-base.tsx

```diff
@@ -0,0 +1,77 @@
+import { autoUpdate, computePosition, flip, offset, Placement } from '@floating-ui/dom';
+import { Component, Host, h, Prop, Event, EventEmitter, Listen, Element } from '@stencil/core';
+import { IDropdownBase, IDropdownBaseEvents } from './dropdown.types';
+
+@Component({
+	tag: 'kv-dropdown-base',
+	styleUrl: 'dropdown-base.scss',
+	shadow: true
+})
+export class KvDropdownBase implements IDropdownBase, IDropdownBaseEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: true }) isOpen?: boolean = false;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) placement?: Placement = 'bottom';
+
+	/** @inheritdoc */
+	@Event() openStateChange: EventEmitter<boolean>;
+
+	@Element() el: HTMLKvDropdownBaseElement;
+
+	@Listen('click', { target: 'window' })
+	checkForClickOutside(ev: { path: HTMLElement[] }) {
+		if (ev.path.some(element => element === this.el)) {
+			return;
+		}
+
+		this.isOpen = false;
+		this.openStateChange.emit(this.isOpen);
+	}
+
+	private closePositionAutoUpdate;
+
+	componentDidRender() {
+		const dropdownAction = this.el.shadowRoot.querySelector('#dropdown-action') as HTMLElement;
+		const dropdownList = this.el.shadowRoot.querySelector('#dropdown-list') as HTMLElement;
+
+		if (this.isOpen) {
+			this.closePositionAutoUpdate = autoUpdate(dropdownAction, dropdownList, () => {
```

Comment:
Only to know, it's unnecessary to destroy the closePositionAutoUpdate on umount?

Reply:
The component itself is not unmounted, only some part of it has disappeared. This is just for tooltip the position to be reseted.